ReactWindows/ReactNative/Views/DatePicker/ReactCalendarDatePickerManager.cs

```diff
@@ -0,0 +1,232 @@
+        /// <param name="day">The day.</param>
+        [ReactProp("firstDayOfWeek")]
+        public void SetFirstDayOfWeek(CalendarDatePicker view, int day)
+        {
+            view.FirstDayOfWeek = (Windows.Globalization.DayOfWeek)day;
+        }
+
+        /// <summary>
+        /// This method returns the <see cref="ReactCalendarDatePickerShadowNode"/>
+        /// which will be then used for measuring the position and size of the
+        /// view.
+        /// </summary>
+        /// <returns>The shadow node instance.</returns>
+        public override ReactCalendarDatePickerShadowNode CreateShadowNodeInstance()
+        {
+            return new ReactCalendarDatePickerShadowNode();
+        }
+
+        /// <summary>
+        /// Implement this method to receive optional extra data enqueued from
+        /// the corresponding instance of <see cref="ReactShadowNode"/> in
+        /// <see cref="ReactShadowNode.OnCollectExtraUpdates"/>.
+        /// </summary>
+        /// <param name="root">The root view.</param>
+        /// <param name="extraData">The extra data.</param>
+        public override void UpdateExtraData(CalendarDatePicker root, object extraData)
+        {
+        }
+
+        /// <summary>
+        /// Unbind event handlers
+        /// </summary>
+        /// <param name="reactContext">The React context.</param>
+        /// <param name="view">The view.</param>
+        public override void OnDropViewInstance(ThemedReactContext reactContext, CalendarDatePicker view)
+        {
+            base.OnDropViewInstance(reactContext, view);
+            view.DateChanged -= OnDateChanged;
+        }
+
+        /// <summary>
+        /// Creates a new view instance of type <see cref="CalendarDatePicker"/>.
+        /// </summary>
+        /// <param name="reactContext">The React context.</param>
+        /// <returns>The view instance.</returns>
+        protected override CalendarDatePicker CreateViewInstance(ThemedReactContext reactContext)
+        {
+            CalendarDatePicker picker = new CalendarDatePicker();
+            return picker;
+        }
+
+        /// <summary>
+        /// Binds the <see cref="CalendarDatePicker.DateChanged"/> event to a handler
+        /// </summary>
+        /// <param name="reactContext">The react context.</param>
+        /// <param name="view">The view instance.</param>
+        protected override void AddEventEmitters(ThemedReactContext reactContext, CalendarDatePicker view)
+        {
+            base.AddEventEmitters(reactContext, view);
+            view.DateChanged += OnDateChanged;
+        }
+
+        /// <summary>
+        /// Date changed update handler
+        /// </summary>
+        /// <param name="sender">The event sender.</param>
+        /// <param name="e">The event.</param>
+        private void OnDateChanged(object sender, CalendarDatePickerDateChangedEventArgs e)
+        {
+            if (!e.NewDate.HasValue)
```

Comment:
Looks like you won't be firing an event if the `NewDate` is set to null. Shouldn't we support an event that allows clearing the date value?

Reply:
Actually, now I don't see how the  `NewDate` can ever be null, as the UI components do not allow clearing the value, so this logic might not even be necessary, but I might have overlooked some option in the pickers. Otherwise sure, an event with a null date can be fired.